py3status/modules/xrandr.py

```diff
@@ -141,17 +156,29 @@ def _set_available_combinations(self):
         available_combinations = set()
         combinations_map = {}
 
+        if self.output_combinations is None:
+            whitelist = None
+        else:
+            whitelist = self.output_combinations.split('|')
+
         self.max_width = 0
-        for output in range(len(self.layout['connected']) + 1):
+        for output in range(1, len(self.layout['connected']) + 1):
             for comb in combinations(self.layout['connected'], output):
-                if comb:
-                    for mode in ['clone', 'extend']:
-                        string = self._get_string_and_set_width(comb, mode)
-                        if len(comb) == 1:
-                            combinations_map[string] = (comb, None)
-                        else:
-                            combinations_map[string] = (comb, mode)
-                        available_combinations.add(string)
+                for mode in ['clone', 'extend']:
+                    string = self._get_string_and_set_width(comb, mode)
+                    if whitelist and string not in whitelist:
+                        continue
+                    if len(comb) == 1:
+                        combinations_map[string] = (comb, None)
+                    else:
+                        combinations_map[string] = (comb, mode)
+                    available_combinations.add(string)
+
+        # Preserve the order in which user defined the output combinations
+        if whitelist is not None:
+            available_combinations = [comb for comb in whitelist
+                                      if comb in available_combinations][::-1]
```

Comment:
I don't really like `[::-1]` as it's not very clear and explicit is good.  Also explaining why we do this might be good
```
available_combinations = [comb for comb in whitelist
    if comb in available_combinations][::-1]
```
so

```
available_combinations = [comb for comb in whitelist
    if comb in available_combinations]
# why are we reversing this?
available_combinations.reverse()
```